arangod/Transaction/Methods.cpp

```diff
@@ -2255,12 +2257,20 @@ Future<OperationResult> transaction::Methods::documentCoordinator(
 Future<OperationResult> transaction::Methods::documentLocal(
     std::string const& collectionName, VPackSlice value,
     OperationOptions options) {
-  auto res = co_await GetDocumentProcessor::create(*this, collectionName, value,
-                                                   options);
-  if (res.fail()) {
-    co_return OperationResult(std::move(res).result(), options);
+  auto fut = Future<OperationResult>::makeEmpty();
+  // Destroy the GetDocumentProcessor before awaiting the future, so the
+  // BuilderLeasers return their builders before the Context might be
+  // gone.
+  {
+    auto getDocumentProcessorRes = co_await GetDocumentProcessor::create(
+        *this, collectionName, value, options);
+    if (getDocumentProcessorRes.fail()) {
+      co_return OperationResult(std::move(getDocumentProcessorRes).result(),
+                                options);
```

Comment:
nit: `options` could be moved here.
```suggestion
      co_return OperationResult(std::move(getDocumentProcessorRes).result(),
                                std::move(options));
```